### PR TITLE
crowbar: Avoid exporting backups in the backup

### PR DESCRIPTION
**Pull request by vuntz**
This just makes the size go exponential.

### CODE REVIEW

**Comment by aplanas** [comment]
+1

**Comment by MaximilianMeister** [comment]
the only good thing to have the backups in the backup would be if I restore in a non upgrade context. because after restore, my backup is gone and that looks a bit weird to the user...

**Comment by tboerger** [comment]
I would prefer to always use rsync.

**Comment by vuntz** [comment]
> the only good thing to have the backups in the backup would be if I restore in a non upgrade context. because after restore, my backup is gone and that looks a bit weird to the user...

Can't we automatically re-import the backup we're restoring from?

**Comment by rsalevsky** [comment]
Should be possible to re-import the current backup. But I would expect that I have all backups after restoring... Not sure if the user expects the same. Maybe we should add a checkbox for including the backups?

**Comment by MaximilianMeister** [comment]
> Can't we automatically re-import the backup we're restoring from?

we would have to add the database entry manually, and not do the `cleanup` at the end of a successful restore. or somehow use the `upload` mechanism?

File: crowbar_framework/lib/crowbar/backup/export.rb [hunk 1]
```diff
@@ -123,7 +123,13 @@ def crowbar
           else
             # copy files with higher permissions
             dest = data_dir.join(destination).to_s
-            system("sudo", "cp", "-a", source, dest)
+            if source == "/var/lib/crowbar"
+              # avoid doing an export of the existing backups, to avoid growing
+              # size of backups
+              system("sudo", "rsync", "-a", "#{source}/", "--exclude", "backup", dest)
+            else
+              system("sudo", "cp", "-a", source, dest)
```
Comment (vuntz) [inline crowbar_framework/lib/crowbar/backup/export.rb]: That's just the old code :-) I can change it to use rsync too.

Comment (nicolasbock) [inline crowbar_framework/lib/crowbar/backup/export.rb]: You would save yourself an `else` :)

Comment (vuntz) [inline crowbar_framework/lib/crowbar/backup/export.rb]: Not really, because we don't want to exclude backup subdirectories in other directories :-)

Comment (nicolasbock) [inline crowbar_framework/lib/crowbar/backup/export.rb]: Well, that's true I suppose.

**Comment by jsuchome** [comment]
+1

**Comment by vuntz** [comment]
@skazi0 @jsuchome did one of you try the patch? It wasn't tested since February, so I'm simply wondering if it still works as intended.

**Comment by jsuchome** [comment]
No, I did not test it :-(

**Comment by vuntz** [comment]
> No, I did not test it :-(

Okay, I'll see if I can test it before merging.

**Comment by vuntz** [comment]
Still works, merging.

**Comment by bmwiedemann** [comment]
was cherry-picked to 3.0 in

```
commit 2d527b4563f9ec0e4eb5d247a317c3dcde329e38
Author:     Vincent Untz <vuntz@suse.com>
AuthorDate: Fri Feb 12 01:35:29 2016 +0100
Commit:     Maximilian Meister <mmeister@suse.de>
CommitDate: Mon Dec 12 16:43:10 2016 +0100
```

**Pull request merged**